### ISSUE TITLE
2018-08-10 patch: exception parsing DEFAULTREALITY.MBIN

### ISSUE THREAD

**Issue by DoctorVanGogh** [monkeyman192/MBINCompiler#48]
```powershell
PS [foo]NMS Mass Tools\EXTRACTED\METADATA\REALITY> & '..\..\..\..\..\MBINCompiler\Build\Debug\MBINCompiler.exe' .\DEFAULTREALITY.MBIN
Values index out of Range. Struct: libMBIN.Models.Structs.GcStatsTypes field: StatsType
ERROR: An unknown exception occurred!
Error
Stacktrace:
   at libMBIN.Models.NMSTemplate.FinishDeserialize() in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 1218
   at libMBIN.Models.NMSTemplate.DeserializeBinaryTemplate(BinaryReader reader, String templateName) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 212
   at libMBIN.Models.NMSTemplate.DeserializeValue(BinaryReader reader, Type field, NMSAttribute settings, Int64 templatePosition, FieldInfo fieldInfo, NMSTemplate parent) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 157
   at libMBIN.Models.NMSTemplate.DeserializeBinaryTemplate(BinaryReader reader, String templateName) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 200
   at libMBIN.Models.NMSTemplate.DeserializeValue(BinaryReader reader, Type field, NMSAttribute settings, Int64 templatePosition, FieldInfo fieldInfo, NMSTemplate parent) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 157
   at libMBIN.Models.NMSTemplate.DeserializeValue(BinaryReader reader, Type field, NMSAttribute settings, Int64 templatePosition, FieldInfo fieldInfo, NMSTemplate parent) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 151
   at libMBIN.Models.NMSTemplate.DeserializeBinaryTemplate(BinaryReader reader, String templateName) in [foo]\MBINCompiler\libMBIN\Source\Models\NMSTemplate.cs:line 200
   at libMBIN.MBINFile.GetData() in [foo]\MBINCompiler\libMBIN\Source\MBINFile.cs:line 71
   at MBINCompiler.Program.DecompileFile(String inputPath, String outputPath, Boolean getVersion, Boolean verbose) in [foo]\MBINCompiler\MBINCompiler\Program.cs:line 49
   at MBINCompiler.Program.Main(String[] args) in [foo]\MBINCompiler\MBINCompiler\Program.cs:line 335
```

They __changed__ something 😱

**Comment by Kriiks** [comment]
Are you using latest Compiler? Works for me with 1.55

**Comment by DoctorVanGogh** [comment]
Dang, I was "still" on 8f945b89b8ad809a83be09b37167854a1e5b9834. 

Alas... no joy with HEAD, now I'm getting `Failed to deserialize template "?????   ", has the structure been mapped yet?`

Did you try with **[today's patch](https://steamdb.info/depot/275851/history/)** and a fresh unpak?

[Affected file](https://drive.google.com/open?id=1SiOhEATv515albAgI2R36iYMPOgZ5leX)

**Comment by Kriiks** [comment]
Yep, that same file works for me with [https://github.com/monkeyman192/MBINCompiler/commit/a2985d8e3346e115c45cfda458c8874833079c2e](url) and 1.55 NMS patch.
I have a problem with NMS_TESTREALITY_GCTECHNOLOGYTABLE.MBIN tho :D

**Comment by monkeyman192** [comment]
We had a few problems with aad2fe50dded095fdaa6c4300347f4f2ff6712b9 which were fixed in 80adc5651d2417cc8f53bb0fc2ecb9983c08cf0f
The latest build should have fixed all these issues.
